### PR TITLE
LIME-343: + fixed routes

### LINKED ISSUE #343: BUG:  Landing page is displayed for a moment to the logged in user when he tries to reach Home page
### Describe your issue

 Landing page is displayed for a moment to the logged in user when he tries to reach Home page  - after displaying loader before displaying Overview page

### Steps to reproduce

1. Go to https://live-in-motion-everyday.com/
2. Go to Sign in page, login with valid credentials
3. Try to reach Home page through URL https://live-in-motion-everyday.com/
4. Pay attention to the page, displayed before redirection to overview page

### What was the actual result?

 Landing page is displayed for a moment to the logged in user when he tries to reach Home page  - after displaying loader before displaying Overview page

### What was the expected result?

Landing page is not displayed logged in user when he tries to reach Home page  - only loader and than Overview page

### Put here any screenshots or videos (optional)

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/123622110/16225f6b-8ba8-4245-94eb-78580024df93

### What OS are you seeing the problem on?

Windows

### What browsers are you seeing the problem on?

Chrome

### Severity

Minor (S4)

### PR DESCRIPTION
**Pull request by ivanbilobrov07**

### CODE REVIEW

File: frontend/src/app/app.tsx [hunk 1]
```diff
@@ -14,12 +14,13 @@ import {
...
-    const [isRefreshing, setIsRefreshing] = useState(true);
+    const [initialRefreshing, setInitialRefreshing] = useState(true);
     const dispatch = useAppDispatch();
     const navigate = useNavigate();
 
-    const { redirectPath } = useAppSelector(({ app }) => ({
+    const { redirectPath, isRefreshing } = useAppSelector(({ app, auth }) => ({
         redirectPath: app.redirectPath,
+        isRefreshing: auth.isRefreshing,
     }));
```
Comment (nikita-remeslov) [inline frontend/src/app/app.tsx]: better to create 2 selectors

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.tsx [hunk 2]
```diff
@@ -38,10 +39,10 @@ const App: React.FC = () => {
             }
         };
 
-        void refreshUser().finally(() => setIsRefreshing(false));
+        void refreshUser().finally(() => setInitialRefreshing(false));
     }, [dispatch]);
 
-    if (isRefreshing) {
+    if (initialRefreshing || isRefreshing) {
```
Comment (dimapopovych) [inline frontend/src/app/app.tsx]: Why do we need two states for this refresh?

Comment (ivanbilobrov07) [reply frontend/src/app/app.tsx]: Before refreshUser() is executed in useEffect, there's an issue that landing page or auth is displayed

**Review by nikita-remeslov — COMMENTED** [review]
why can't we wait until the user is loaded and only after that show layout?

**Comment by stpalamar** [comment]
I guess I have simpler solution.
We can just make isRefreshing true in initialState and set it false if there is no token



image

And keep this redirect in auth and landing pages if we have user
image

File: frontend/src/bundles/landing/pages/landing.tsx [hunk 3]
```diff
@@ -37,16 +36,10 @@ import {
 import { FEATURES, TESTIMONIALS } from '../constants/constants.js';
 
 const Landing = (): JSX.Element => {
-    const { theme } = useAppSelector(({ theme }) => theme);
-    const { user, isRefreshing } = useAppSelector(({ auth }) => auth);
+    const { theme } = useAppSelector((state) => state.theme);
```
Comment (stpalamar) [inline frontend/src/bundles/landing/pages/landing.tsx]:
```suggestion
    const { theme } = useAppSelector(({ theme }) => theme);
```

**Review by stpalamar — CHANGES_REQUESTED** [review]

**Review by stpalamar — DISMISSED** [review]

**Review by SofiiaTrokhymchuk — DISMISSED** [review]

**Review by nikita-remeslov — DISMISSED** [review]

**Pull request merged**